__tests__/app.test.js

```diff
@@ -272,3 +273,55 @@ describe("GET api/articles/:article_id (comment count)", () => {
     });
   })
 })
+
+describe("GET /api/articles", () => {
+  describe("STATUS: 200", () => {
+    test("endpoint responds with 200 status and an array of all article objects", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { allArticles } = body;
+          expect(allArticles).toBeInstanceOf(Array);
+          expect(allArticles[0]).toBeInstanceOf(Object);
+          expect(allArticles).toHaveLength(12);
+        });
+    });
+    test("endpoint responds with an array of article objects, which include specified keys", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { allArticles } = body;
+          allArticles.forEach((article) => {
+            expect(article.hasOwnProperty("username")).toBe(true);
+            expect(article.hasOwnProperty("title")).toBe(true);
+            expect(article.hasOwnProperty("article_id")).toBe(true);
+            expect(article.hasOwnProperty("topic")).toBe(true);
+            expect(article.hasOwnProperty("created_at")).toBe(true);
+            expect(article.hasOwnProperty("votes")).toBe(true);
+            expect(article.hasOwnProperty("comment_count")).toBe(true);
+          });
+        });
+    });
+    test("article objects within the array should be sorted by creation day", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { allArticles } = body;
+          expect(allArticles).toBeSortedBy('created_at', {descending: true});
+       })
+    })
+  });
+  describe("STATUS: 404", () => {
+    test(" responds with a status code 404 if endpoint is incorrect/spelling erros", () => {
+      return request(app)
+        .get("/api/article")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Not Found!");
+        });
+    });
+  });
+});
```

Comment:
This is a generic test which is effectively checking /api/notanedpoint - this should really be a stand alone test. Remove any duplicate tests from other endpoints.

Reply:
@corkesw dublicates removed, test moved to be a stand alone test. Am I ok to merge?